app/modes/library/agent.py

```
@@ -0,0 +1,490 @@
+        else:
+            return {
+                'status': 'completed',
+                'message': 'Already generated',
+                'book_id': book_id,
+                'total_chapters': total_chapters_actual,
+                'completed_chapters': completed_chapters_actual
+            }
+
+    # If book is actually complete but progress says otherwise, fix it
+    if not needs_generation and progress.status != 'completed':
+        logger.info(f"Book {book_id} is complete but status was {progress.status}, fixing")
```

Ruff (0.15.1)

[warning] Logging statement uses f-string

(G004)